src/test/java/com/ita/if103java/ims/dao/DashboardDaoImplTest.java

```diff
@@ -0,0 +1,225 @@
+import org.springframework.data.domain.Sort;
+import org.springframework.jdbc.core.JdbcTemplate;
+import java.util.ArrayList;
+
+import static com.ita.if103java.ims.util.JDBCUtils.getOrder;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class DashboardDaoImplTest {
+
+    @Mock
+    private WarehouseLoadRowMapper warehouseLoadRowMapper;
+    @Mock
+    private PopularItemsRowMapper popularItemsRowMapper;
+    @Mock
+    private EndingItemsRowMapper endingItemsRowMapper;
+    @Mock
+    private WarehousePremiumStructRowMapper warehousePremiumStructRowMapper;
+    @Mock
+    private ChargeCapacityRowMapper chargeCapacityRowMapper;
+    @Mock
+    JdbcTemplate jdbcTemplate;
+
+    @InjectMocks
+    DashboardDaoImpl dashboardDaoImpl;
+
+    @BeforeEach
+    public void setUp() {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    public void testWarehouseLoad(){
+        dashboardDaoImpl.findWarehouseLoadByAccountId(1L);
+        verify(jdbcTemplate, times(1))
+            .query(
+                DashboardDaoImpl.Queries.SQL_FIND_WAREHOUSE_LOAD_BY_ACCOUNT_ID, warehouseLoadRowMapper, 1L
+            );
+    }
+
+    @Test
+    public void testWarehouseLoadException1(){
+        when(jdbcTemplate.query(
+            DashboardDaoImpl.Queries.SQL_FIND_WAREHOUSE_LOAD_BY_ACCOUNT_ID, warehouseLoadRowMapper, -1L))
+            .thenThrow(EmptyResultDataAccessException.class);
+        assertThrows(DashboardDataNotFoundException.class, () -> dashboardDaoImpl.findWarehouseLoadByAccountId(-1L));
+
+    }
+
+    @Test
+    public void testPopularItems1() {
+        PopularItemsRequestDto popularItems =
+            new PopularItemsRequestDto(3, DateType.YEAR, PopType.TOP, "17-02-2020");
+
+        dashboardDaoImpl.findPopularItems(popularItems, 1L);
+
+        verify(jdbcTemplate, times(1))
+            .query(
+                DashboardDaoImpl.Queries.SQL_FIND_POPULAR_ITEMS + DashboardDaoImpl.Queries.SQL_POP_YEAR +
+                    DashboardDaoImpl.Queries.SQL_ATR_POP,
+                popularItemsRowMapper,
+                1L, popularItems.getDate(), popularItems.getQuantity()
+            );
+    }
+
+    @Test
+    public void testPopularItems2() {
+        PopularItemsRequestDto popularItems =
+            new PopularItemsRequestDto(9, DateType.MONTH, PopType.BOT, "17-02-2019");
+
+        dashboardDaoImpl.findPopularItems(popularItems, 1L);
+
+        verify(jdbcTemplate, times(1))
+            .query(
+                DashboardDaoImpl.Queries.SQL_FIND_POPULAR_ITEMS +
+                    DashboardDaoImpl.Queries.SQL_POP_MONTH + DashboardDaoImpl.Queries.SQL_ATR_UNPOP,
+                popularItemsRowMapper,
+                1L, popularItems.getDate(), popularItems.getDate(), popularItems.getQuantity()
+            );
+    }
+
+
+    @Test
+    public void testPopularItems3() {
```

Comment:
better to name
testPopularItems_unpopularDuringWholeTime

And so on and so forth